lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnDriverRuntimeRestartManager.java

```diff
@@ -78,31 +78,44 @@ private YarnDriverRuntimeRestartManager(@Parameter(YarnEvaluatorPreserver.class)
   }
 
   /**
-   * Determines whether the application master has been restarted based on the container ID environment
+   * Determines the number of times the Driver has been submitted based on the container ID environment
    * variable provided by YARN. If that fails, determine whether the application master is a restart
-   * based on the number of previous containers reported by YARN.
-   * @return true if the application master is a restarted instance, false otherwise.
+   * based on the number of previous containers reported by YARN. In the failure scenario, returns 1 if restart, 0
+   * otherwise.
+   * @return > 0 if the application master is a restarted instance, 0 otherwise.
    */
   @Override
-  public boolean hasRestarted() {
+  public int getResubmissionAttempts() {
     final String containerIdString = getContainerIdString();
 
     if (containerIdString == null) {
       // container id should always be set in the env by the framework
       LOG.log(Level.WARNING, "Container ID is null, determining restart based on previous containers.");
-      return this.isRestartByPreviousContainers();
+      if (this.isRestartByPreviousContainers()) {
+        LOG.log(Level.WARNING, "Driver is a restarted instance. Returning 1.");
+        return 1;
```

Comment:
Yes, we already do. This is evidenced by the final `return appAttempt -1`. This is only used as a fallback if we fail to get the `appAttempt` from the ResourceManager. Since we ourselves cannot determine how many times the Driver has been restarted, we simply return 1 as a default. It's probably a good idea for us to define a constant `String` that allows better interpretation though. I will update it.